File: dotnet/Samples/LanguageDetection/Text/EdgeTextLanguageDetectionService.cs
using LanguageWeaver.Sdk.Common.Edge;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.LanguageDetection.Edge.Request;

namespace LanguageWeaver.Sdk.Samples.LanguageDetection.Text;

public class EdgeLanguageDetectionService
{
    public static void Main()
    {
        using var lwClient = new EdgeLanguageWeaverClient().Build();
        var edgeTextLanguageDetectionRequest = new EdgeTextLanguageDetectionRequest()
        {
            Input = "A mountain is generally steeper than a hill",
            InputFormat = Format.Plain,
            Encoding = "UTF-8"
        };

        var edgeLanguageDetectionResult = lwClient.DetectLanguage(edgeTextLanguageDetectionRequest);

        edgeLanguageDetectionResult.Languages
            .ForEach(language => Console.WriteLine(language.Name));
    }
}

File: javascript/backend/samples/language-detection/text/EdgeTextLanguageDetectionService.js
const {
    EdgeLanguageWeaverClient,
    EdgeTextLanguageDetectionRequest,
    Format,
} = require("@language-weaver/lw-sdk-js");
const detectEdgeTextLanguage = async () => {
    try{
        const edgeLanguageWeaverClient = await new EdgeLanguageWeaverClient().build();
        const edgeTextLanguageDetectionRequest = new EdgeTextLanguageDetectionRequest();
        edgeTextLanguageDetectionRequest.input = "A mountain is generally steeper than a hill";
        edgeTextLanguageDetectionRequest.inputFormat = Format.PLAIN;

        const edgeTextLanguageDetectionResult = await edgeLanguageWeaverClient.detectEdgeTextLanguage(edgeTextLanguageDetectionRequest);
        console.log(edgeTextLanguageDetectionResult);
    } catch (e) {
        console.log(e);
    }
}

detectEdgeTextLanguage();